identity/src/main/java/com/stripe/android/identity/viewmodel/IdentityViewModel.kt

```diff
@@ -726,14 +736,26 @@ internal class IdentityViewModel constructor(
         APIException::class
     )
     suspend fun postVerificationPageData(
-        collectedDataParam: CollectedDataParam,
-        clearDataParam: ClearDataParam
-    ) =
+        collectedDataParam: CollectedDataParam
+    ): VerificationPageData {
         identityRepository.postVerificationPageData(
             verificationArgs.verificationSessionId,
             verificationArgs.ephemeralKeySecret,
             collectedDataParam,
-            clearDataParam
+            calculateClearDataParam(collectedDataParam)
+        ).let {
+            _collectedData.update { oldValue ->
+                oldValue.mergeWith(collectedDataParam)
+            }
+            return it
+        }
+    }
+
+    private fun calculateClearDataParam(dataToBeCollected: CollectedDataParam) =
```

Comment:
Instead of passing hardcoded ClearDataParam, we can calculate it here by
(allTypes-collectedTypes-typeToBeCollected)